src/idiomatic/leveraging-the-type-system/token-types/branded-02-phantomdata.md

```diff
@@ -0,0 +1,124 @@
+---
+minutes: 20
+---
+
+# `PhantomData` and Lifetime Subtyping (Branding 2/4)
+
+We need a mechanism to make lifetimes distinct enough from each other to not
+compile when we try to compare them.
+
+<!-- dprint-ignore-start -->
+```rust,editable
+use std::marker::PhantomData;
+
+#[derive(Default)]
+struct InvariantLifetime<'id>(PhantomData<&'id ()>); // The main focus
+
+struct Wrapper<'a> { value: u8, invariant: InvariantLifetime<'a> }
+
+fn lifetime_separator<T>(value: u8, f: impl for<'a> FnOnce(Wrapper<'a>) -> T) -> T {
+    f(Wrapper { value, invariant: InvariantLifetime::default() })
+}
+
+fn compare_lifetimes<'a>(left: Wrapper<'a>, right: Wrapper<'a>) {}
+
+fn main() {
+    lifetime_separator(1, |wrapped_1| {
+        lifetime_separator(2, |wrapped_2| {
+            // We want this to NOT compile
+            compare_lifetimes(wrapped_1, wrapped_2);
+        });
+    });
+}
+```
+<!-- dprint-ignore-end -->
+
+<details>
+
+<!-- TODO: Link back to PhantomData in the borrowck invariants chapter.
+- We saw `PhantomData` back in the Borrow Checker Invariants chapter.
+-->
+
+- **Goal**: We want two lifetimes that the rust compiler cannot determine if one
```

Comment:
Maybe step back a little and say that Rust defines a subtyping relationship between references, and references with longer lifetimes implicitly convert into shorter lifetimes. This also happens when a user-defined type has a lifetime parameter, Rust defines a subtyping relationship.

I know you're talking more about subtyping later, but maybe a brief mention upfront would be good. I would like you to consider how you'd introduce the flow from the idea ("Use a lifetime as a unique brand for each token") to the suggested solution (disable subtyping) in the shortest way possible when first introducing the slide.

... thus our goal: eliminate that subtying relationship, in order for each Token's lifetime to be unique and incompatible with every other Token.